torch/_dynamo/utils.py

```diff
@@ -4687,7 +4687,50 @@ def _extract_tensor_dict(t: torch.Tensor) -> dict[str, Any]:
 # e.g. torch.cuda.Event is a prime example.
 user_obj_id_to_weakref: dict[int, weakref.ReferenceType[object]] = {}
 
+# Similar to above, except instead of ID, we use a dynamo-generated index
```

Comment:
Consider moving all of this to a separate file, where we can add these sync points